src/util/system.cpp

```diff
@@ -1068,8 +1068,8 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
     }
 
     // If datadir is changed in .conf file:
-    gArgs.ClearPathCache();
```

Comment:
nit: Looks like this shouldn't have been called on `gArgs` regardless of this PR, perhaps useful to split out in separate commit? In the 3 callsites of `ArgsManager::ReadConfigFiles` I see, it is only called on `gArgs` anyway, so I don't see any actual behaviour change/potential bug because of it.